src/main/java/baritone/utils/GuiClick.java

```diff
@@ -93,10 +92,16 @@ protected void mouseReleased(int mouseX, int mouseY, int mouseButton) {
                 Helper.HELPER.logDirect(component);
                 clickStart = null;
             } else {
-                BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalTwoBlocks(currentMouseOver));
+                if (currentMouseOver != null)//Catch this, or else a click into void will result in a crash
+                    BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalBlock(currentMouseOver));
+                else
+                    Helper.HELPER.logDirect("Sorry, I can't go to nothing");
```

Comment:
no need to notify user

Reply:
This can still crash if someone drags the mouse so you might as well just check it once for the entire function. https://github.com/cabaletta/baritone/blob/0055a059f71be48d68d6ad92aaad1712dee3be7a/src/main/java/baritone/utils/GuiClick.java#L84

Reply:
So I should remove the message, and add the if-clause around everything?
Like this?:
```
if (currentMouseOver != null) {
    if (mouseButton == 0) {
       ...
    } else if (mouseButton == 1) {
        ...
    }
}
clickStart = null;
```

Reply:
> So I should remove the message, and add the if-clause around everything?
> Like this?:
> 
> ```
> if (currentMouseOver != null) {
>     if (mouseButton == 0) {
>        ...
>     } else if (mouseButton == 1) {
>         ...
>     }
> }
> clickStart = null;
> ```

Looks good to me